test/logging/sink.js

```diff
@@ -0,0 +1,200 @@
+var mockery = require('mockery');
+var nodeutil = require('util');
+
+var ServiceObject = require('../../lib/common/service-object.js');
+var util = require('../../lib/common/util.js');
+
+function FakeServiceObject() {
+  this.calledWith_ = arguments;
+  ServiceObject.apply(this, arguments);
+}
+
+nodeutil.inherits(FakeServiceObject, ServiceObject);
+
+describe('Sink', function() {
+  var Sink;
+  var sink;
+
+  var LOGGING = {
+    createSink: util.noop
+  };
+  var SINK_NAME = 'sink-name';
+
+  before(function() {
+    mockery.registerMock('../common/service-object.js', FakeServiceObject);
+
+    mockery.enable({
+      useCleanCache: true,
+      warnOnUnregistered: false
+    });
+
+    Sink = require('../../lib/logging/sink.js');
+  });
+
+  after(function() {
+    mockery.deregisterAll();
+    mockery.disable();
+  });
+
+  beforeEach(function() {
+    sink = new Sink(LOGGING, SINK_NAME);
+  });
+
+  describe('instantiation', function() {
+    it('should inherit from ServiceObject', function(done) {
+      var loggingInstance = extend({}, LOGGING, {
+        createSink: {
+          bind: function(context) {
+            assert.strictEqual(context, loggingInstance);
+            done();
+          }
+        }
+      });
+
+      var sink = new Sink(loggingInstance, SINK_NAME);
+      assert(sink instanceof ServiceObject);
+
+      var calledWith = sink.calledWith_[0];
+
+      assert.strictEqual(calledWith.parent, loggingInstance);
+      assert.strictEqual(calledWith.baseUrl, '/sinks');
+      assert.strictEqual(calledWith.id, SINK_NAME);
+      assert.deepEqual(calledWith.methods, {
+        create: true,
+        delete: true,
+        getMetadata: true
+      });
+    });
+
+    it('should localize the name', function() {
+      assert.strictEqual(sink.name, SINK_NAME);
+    });
+  });
+
+  describe('setFilter', function() {
+    var FILTER = 'filter';
+
+    it('should call set metadata', function(done) {
+      sink.setMetadata = function(metadata, callback) {
+        assert.strictEqual(metadata.filter, FILTER);
+        callback(); // done()
+      };
+
+      sink.setFilter(FILTER, done);
+    });
+  });
+
+  describe('setMetadata', function() {
+    var METADATA = { a: 'b', c: 'd' };
+
+    it('should refresh the metadata', function(done) {
+      sink.getMetadata = function() {
+        done();
+      };
+
+      sink.setMetadata(METADATA, assert.ifError);
+    });
+
+    it('should exec callback with error from refresh', function(done) {
+      var error = new Error('Error.');
+      var apiResponse = {};
+
+      sink.getMetadata = function(callback) {
+        callback(error, null, apiResponse);
+      };
+
+      sink.setMetadata(METADATA, function(err, apiResponse_) {
+        assert.strictEqual(err, error);
+        assert.strictEqual(apiResponse_, apiResponse);
+        done();
+      });
+    });
+
+    it('should make the correct request', function(done) {
+      var currentMetadata = { e: 'f', g: 'h' };
```

Comment:
Thanks for the comments above!
In which case, maybe change currentMetadata to e.g. { 'c': 'e', 'g': 'h'} so the intersection of current and new metadata is not empty